### PR TITLE
Fix #2668: Recalculate center points when NTP asset come in.

### LINKED ISSUE #2668: SR image on first tab is not centered, auto fixes when tab focus is lost [Follow up to #2664]
### Description: 
SR image on first tab is not centered, auto fixes when tab focus is lost [Follow up to #2664]

### Steps to Reproduce 
  1. Visit a SR link
  2. Install 1.18(20.06.24.15)
  3. First NTP loads asset but is not centre aligned
  4. Open a new tab and switch back to first NTP image snaps back into position
4.1. Alternate to step 4 change orientation on first NTP and image snaps back to position

**Actual result:** 
https://bravesoftware.slack.com/archives/C06UXF3KJ/p1593153382082300

**Expected result:**
All tabs should load the image correctly 

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.18(20.06.24.15)

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)? NA

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #2668 
ref #2664 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift [hunk 1]
```diff
@@ -334,6 +311,39 @@ class NewTabPageViewController: UIViewController, Themeable {
...
         }
+        
+        // This is usually done in `viewDidLayoutSubviews`, one exception:
+        // First launch, intial image loads, NTP assets is being downloaded,
+        // after it downloods we replace current image with sponsored one
+        // and have to recalculate center points.
+        calculateCenterPoints()
+    }
+    
+    private func calculateCenterPoints() {
```
Comment (kylehickinson) [inline Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift]: Can you rename this to `calculateBackgroundCenterPoints`

**Review by kylehickinson — APPROVED** [review]
Just a minor thing if you can fix before merging.

How come this fix works? (I mean why does adding this to `traitCollectionDidChange` affect backgrounds that are downloaded later)

**Comment by iccub** [comment]
It's added to `setBackgroundImage` method, github showing wrong

Basically we do not call layout-update code when new image is set.
And when first image doesn't use focal points, and second does use them, this resulted in bad behavior

**Comment by kylehickinson** [comment]
Ahh, gotcha, GitHub tricked me lol.

This makes sense, thanks.

**Pull request merged**